### PR TITLE
#1366 "CVars work by necessarily-equal" + fixes

### PR DESCRIPTION
**Pull request by slipher**
Fixes on top of the #1366 changes:
- Remove unneeded callbacks from BP vars
- Comma splitter: Use standard library std::string functionality instead of writing own
- Fix misspelled item names
- Don't use "allowed" as argument name for forbidden things
- Don't use vectors in 'conditions' struct in gentity_t because these things get memset
- Improve bot cvar reading

### CODE REVIEW

File: src/shared/parse.cpp [hunk 1]
```diff
@@ -4276,3 +4276,38 @@ int Parse_SourceFileAndLine( int handle, char (&filename)[ MAX_QPATH ], int *lin
...
+see the header file for the documentation
+===============
+*/
+
+static constexpr char csv_delimiters[] = ", ";
+
+Parse_WordListSplitter::Parse_WordListSplitter(std::string str)
+	: _string(std::move(str)), _stop(0)
+{
+	++*this;
```
Comment (necessarily-equal) [inline src/shared/parse.cpp:4293]: neat

File: src/shared/bg_misc.cpp [hunk 2]
```diff
@@ -2313,256 +2314,127 @@ int BG_UnpackEntityNumbers( entityState_t *es, int *entityNums, unsigned int cou
...
-		{
-			p++;
-			q = p;
+		if (result != PCL_NONE) {
+			results.push_back(result);
 		}
 		else
-		{
-			break;
+	       	{
```
Comment (necessarily-equal) [inline src/shared/bg_misc.cpp:2373]: indentation went wrong here

**Comment by necessarily-equal** [comment]
there is a regression in your version though:
While `g_disabledEquipment "lcannon"` works and `g_disabledEquipment "gren"` works, `g_disabledEquipment "lcannon,gren"` doesn't.

**Comment by slipher** [comment]
> 
> 
> there is a regression in your version though:
> While `g_disabledEquipment "lcannon"` works and `g_disabledEquipment "gren"` works, `g_disabledEquipment "lcannon,gren"` doesn't.

Fixed

**Pull request merged**